src/js/components/Header/Header.js

```diff
@@ -1,19 +1,83 @@
-import React from 'react';
-
+import React, { useContext, useEffect, useState } from 'react';
+import { ThemeContext } from 'styled-components';
+import { useForwardedRef } from '../../utils';
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const containerRef = useForwardedRef(ref);
+  const [stickyStyles, setStickyStyles] = useState();
+
+  useEffect(() => {
+    // threshold inital value set to 0
+    const threshold = 0;
+    // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    // run our event listener callback once in each requestAnimationFrame
+    let runEventListener = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        runEventListener = false;
+        return;
+      }
+      if (scrollY === 0) {
+        runEventListener = false;
+        setStickyStyles(undefined);
+      }
+      if (scrollY < lastScrollY) {
+        setStickyStyles({
+          width: `${containerRef.current.getBoundingClientRect().width}px`,
+          height: `${containerRef.current.getBoundingClientRect().height}px`,
+          left: `${containerRef.current.getBoundingClientRect().left}px`,
+          right: `${containerRef.current.getBoundingClientRect().right}px`,
+          zIndex: `${theme.header?.sticky?.zIndex}`,
+          position: 'fixed',
+          top: '0',
+        });
+      } else setStickyStyles(undefined);
+      lastScrollY = scrollY > 0 ? scrollY : 0;
+      runEventListener = false;
+    };
+
+    const onScroll = () => {
+      if (!runEventListener) {
+        // calculate the new offset after the page gets rendered
+        window.requestAnimationFrame(updateScrollDir);
+        runEventListener = true;
+      }
+    };
+
+    window.addEventListener('scroll', onScroll);
+
+    return () => window.removeEventListener('scroll', onScroll);
+  }, [containerRef, stickyStyles, theme.header?.sticky?.zIndex]);
+
+  return (
+    <>
+      {sticky === 'scrollup' && (
+        <Box
+          height={stickyStyles?.height}
+          width={stickyStyles && stickyStyles.width}
+        />
+      )}
```

Comment:
What is this providing for us?

Reply:
Eric had mentioned having `Box` in the dom 
I think we still need to render a Box at the right height when showing the sticky header as a fixed DOM element.